### PR TITLE
[PDS-146501] Cast DEFAULT_TARGETED_SWEEP_SHARDS=1 down into the briny deeps

### PR DESCRIPTION
**Pull request by jeremyk-91**
**Goals (and why)**:
- Avoid tech debt that led to much (probably unnecessary) angst in a P0

**Implementation Description (bullets)**:
- Don't have misleading constants: rename the old `DEFAULT_TARGETED_SWEEP_SHARDS = 1` to `LEGACY_DEFAULT_TARGETED_SWEEP_SHARDS`, and set `DEFAULT_TARGETED_SWEEP_SHARDS = 8`
- There are likely to be substantially more complex semantics involved with just changing the number.

**Testing (What was existing testing like?  What have you done to improve it?)**:
- Not much.

**Concerns (what feedback would you like?)**:
- Nothing in particular.

**Where should we start reviewing?**: AtlasDbConstants

**Priority (whenever / two weeks / yesterday)**: Yesterday. The assumption drawn from this constant caused a lot of angst in a P0 yesterday.

### CODE REVIEW

File: atlasdb-client/src/main/java/com/palantir/atlasdb/AtlasDbConstants.java
Comment (Jolyon-S) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/AtlasDbConstants.java:154]: Can we actually just bump this to 16?

**Review by Jolyon-S — COMMENTED** [review]
I don't know if you want to wait for @gmaretic's review before merging, but LGTM from my end - just one comment.

**Review by gmaretic — APPROVED** [review]
Not opposed to changing the default to 12 or 16, but this looks good.

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/ShardProgress.java
Comment (gmaretic) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/ShardProgress.java:57]: 👍  for the defensiveness

**Comment by jeremyk-91** [comment]
Let's go to 16, though in this case we _do_ want changelog I think

**Pull request merged**
